src/subprocess.h

```diff
@@ -96,16 +115,24 @@ struct SubprocessSet {
   static HANDLE ioport_;
 #else
   static void SetInterruptedFlag(int signum);
-  static void HandlePendingInterruption();
+  static void SigChldHandler(int signo, siginfo_t* info, void* context);
+
   /// Store the signal number that causes the interruption.
   /// 0 if not interruption.
-  static int interrupted_;
-
+  static volatile int interrupted_;
```

Comment:
Why did you make this volatile?

Reply:
I will make it your call whether `volatile` here stays or goes away.

Reply:
The best option is probably to use [sig_atomic_t](https://en.cppreference.com/w/c/program/sig_atomic_t) which was designed specially for this exact use case.